apps/expo/src/screens/home/SearchDeviceScreen.tsx

```diff
@@ -51,93 +51,125 @@ export default function SearchDeviceScreen({ navigation, route }: SearchDeviceSc
+  //     {
+  //       text: t("screens.home_stack.search_device.open_settings"),
+  //       onPress: () => {
+  //         // eslint-disable-next-line node/handle-callback-err, @typescript-eslint/no-empty-function
+  //         openSettings().catch(e => {});
+  //       },
+  //     },
+  //   ]);
+  // };
 
-    Alert.alert("Error", err, [
-      Platform.OS === "ios"
-        ? {
-          text: "OK",
+  // const onEnableBluetoothError = (err: string) => {
+  //   console.log("onEnableBluetoothError", err);
+
+  //   Alert.alert("Error", err, [
+  //     Platform.OS === "ios"
+  //       ? {
+  //           text: "OK",
+  //         }
+  //       : {
+  //           text: "Open settings",
+  //           onPress: () => {
+  //             // eslint-disable-next-line node/handle-callback-err, @typescript-eslint/no-empty-function
+  //             BluetoothStateManager.openSettings().catch(e => {});
+  //           },
+  //         },
+  //   ]);
+  // };
+
+  // const askForBluetoothPermission = async (): Promise<null> => {
+  //   return new Promise((resolve, reject) => {
+  //     Alert.alert(
+  //       t("screens.home_stack.search_device.bluetooth.alert.title"),
+  //       t("screens.home_stack.search_device.bluetooth.alert.message"),
+  //       [
+  //         {
+  //           text: t("screens.home_stack.search_device.bluetooth.alert.button"),
+  //           onPress: async () => {
+  //             try {
+  //               await requestBluetoothPermission();
+  //               resolve(null);
+  //             } catch (err: unknown) {
+  //               const errorMsg =
+  //                 err instanceof Error
+  //                   ? err.message
+  //                   : t("screens.home_stack.search_device.errors.bluetooth.request_failed");
+  //               onRequestPermissionError(errorMsg);
+
+  //               setIsScanning(false);
+  //               setIsError(true);
+  //               reject(err);
+  //             }
+  //           },
+  //         },
+  //       ]
+  //     );
+  //   });
+  // };
+
+  useEffect(() => {
+    const checkAndEnableBluetooth = async () => {
+      try {
+        const bluetoothState = await BluetoothStateManager.getState();
+        if (bluetoothState !== "PoweredOn") {
+          Alert.alert(
+            t("screens.home_stack.search_device.bluetooth.alert.switch_on"),
+            t("screens.home_stack.search_device.bluetooth.alert.disabled_bluetooth"),
+            [
+              {
+                text: t("screens.home_stack.search_device.bluetooth.alert.cancel"),
+                onPress: () => console.log(t("screens.home_stack.search_device.bluetooth.error.cancel_message")),
+                //style: "cancel",
+              },
+              {
+                text: t("screens.home_stack.search_device.bluetooth.alert.switch_on"),
+                onPress: async () => {
+                  try {
+                    const success = await BluetoothStateManager.openSettings();
```

Comment:
Deze lijkt ook voor iOS te werken dus ik ga er van uit dat het prima is. Volgende week testen als we sudo hebben